### ISSUE TITLE
Issues with --base_weights, --train_connectors and masked training seems to be always active for video.

### ISSUE THREAD

**Issue by mcmanus82** [AkaneTendo25/musubi-tuner#92]
### 1. Masked training seems to be always active for video:
I don't now if this is an LTX thing but the documentation explicitly states that if there is no masking items in the dataset, then these should not be logged.
When training in ltx_mode=video and targe=t2v, video_loss_unmasked_, video_loss_masked and video_mask_active is logged.
video_mask_active is always 0.875 and current_loss = video_loss_masked, which seems to indicate that the loss is indeed reduced.
Dataset config and the training command does not include any masking related data (i never even tried masked training)
When running training that includes audio as well (av) Audio related masking data is also logged, but the activation for that seems to be 1.0 and the current audio loss equals the unmasked audio loss.
<img width="1373" height="377" alt="Image" src="https://github.com/user-attachments/assets/1e7ffa8e-0519-430b-b32b-fd97f58fd009" />

### 2. Using a DoRa for --base_weights results in loss above 2.0 from step 1:
I found this before the current large refactor update, but i did not verify that it still exists. Training goes for about 50 steps than crashes.
The DoRa used is in native musubi format (not comfyui). There are no errors in the terminal and it states that the weights were applied successfully. ltx_mode was video for both the DoRa and the training that was to be started. The training to be started was also run with "use_dora=true"
Merging the same DoRa file into the base model worked fine, checking that model in comfyui showed that it contains the function of the merged DoRa. Starting training with that merged base DiT also worked fine(not using --base_weights).

### 3. --train_connectors does not work with ltx_mode = video:
I forgot to save the logs for this one, but from what i could discern, audio connectors are included even when that is not in the scope of training, and that results in a dimension count mismatch.

If you need more information or logs for any of these, please let me know.
Thank You for the great work :)

**Comment by AkaneTendo25** [comment]
Thank you, I will check this.

**Comment by mcmanus82** [comment]
I figured i would do an experiment for the masking issue. Used a python script to create full white png images with matching resolution and filenames for the training dataset, but not for the validation dataset.
For the 1x1 aspect ratio videos that are used with 512x512 resolution (57 frame for all of them) ones in the training dataset with the full white mask had a cached latent file size of 520.3 Kb while the ones in the validation dataset (no white mask) had a size of 512.2 Kb

I started training, and the video_mask_active value keeps bouncing from 0.875 to 1.0 which is kinda weird :D
<img width="1294" height="332" alt="Image" src="https://github.com/user-attachments/assets/6de1f558-f72f-4b16-94b6-0a40beadde17" />

**Comment by mcmanus82** [comment]
Some additional information / observations:

- Converted the entire dataset from mp4 - h264 to mov ProRes Balanced, results were identical
- When max_frames = 1 and no white masks are used, no mask related data is logged (expected behavior)
- When max_frames = 1 with white masks defined, mv_act is logged with a constant 1.0 (again, expected behavior)
- When max_frames > 9 and no white masks are used, only certain steps have mask related data logged (1,5,13,25,43,44,51,52)
- When max_frames > 9 with white masks defined, the steps that previously had masking data logged (1,5,13,25,43,44,51,52) retain there mv_act values (0.875) while all other steps are logged with mv_act of 1.0
- Increasing batch size from 1 to 2, adds a new second value for mv_act between the previous one and 1.0 (mv_act 0.875 / 0.9375 / 1.0) for some of the steps that had masking data logged (1,5,13,25).
- The higher the max_frames value, the higher the minimum of mv_act is (see table below)
1 frame - mv_act 1.0
9 frames - mv_act 0.5 / 0.75 / 1.0
25 frames - mv_act 0.75 / 0.875 1.0
49 frames - mv_act 0.8571 / 0.9286 / 1.0
57 frames - mv_act 0.875 / 0.9375 / 1.0

- I noticed there is a debug_dataset property when caching / loading datasets, but i was unable to find a way to enable it.
- I tried --debug_mode video for latent caching, and as far as i can tell the videos it produced were good.

**Comment by mcmanus82** [comment]
Adding "--ltx2_first_frame_conditioning_p 0.0" to the training command fixes the issue.
It seems that for some reason first frame conditioning is always active with a certain chance. This would be in line with the mv_act increasing when the frame count is larger, as the first frame would be a smaller 'masked' part of the longer video.

<img width="1367" height="404" alt="Image" src="https://github.com/user-attachments/assets/365c8adc-97c1-4314-898e-2fd80129309e" />

**Comment by AkaneTendo25** [comment]
Thanks for detailed observations. I reviewed each issue, here is what I found:

> `mv_act` / `video_mask_active` without dataset masks

This seems to be expected behavior from first-frame conditioning, not dataset masked-loss training. Since `--ltx2_first_frame_conditioning_p` defaults to `0.1`, when it fires, frame 0 is kept clean and excluded from loss, so mask metrics are logged (as if frame was masked). So it is not a bug, but behavior was easy to misinterpret.
I added docs/troubleshooting clarification.
For pure T2V/no first-frame conditioning, set `--ltx2_first_frame_conditioning_p 0.0`.

> DoRA as `--base_weights`

I could not reproduce it on the current 'ltx-2-dev' branch. The current path correctly detects native DoRA weights. Maybe this error was mitigated naturally as result of some other code changes. 

> `--train_connectors` with `ltx_mode=video`

This should be fixed now. Previously cached `audio_features` could be forwarded in video mode and trigger audio connector handling, now they are now forwarded only for `av`/`audio` modes.

> `debug_dataset`

I added `--debug_dataset` for LTX training, should be usable now.
